substrate/frame/staking/ah-client/src/lib.rs

```diff
@@ -0,0 +1,122 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! TODO
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+pub use pallet::*;
+
+/// The balance type of this pallet.
+pub type BalanceOf<T> = <T as Config>::CurrencyBalance;
+
+#[frame_support::pallet(dev_mode)]
+pub mod pallet {
+	use crate::BalanceOf;
+	use frame_support::pallet_prelude::*;
+	use frame_system::pallet_prelude::*;
+	use pallet_session::historical;
+	use sp_staking::{Exposure, SessionIndex};
+
+	const STORAGE_VERSION: StorageVersion = StorageVersion::new(1);
+	#[pallet::pallet]
+	#[pallet::storage_version(STORAGE_VERSION)]
+	pub struct Pallet<T>(_);
+
+	// todo:
+	// Storage item for new_validator_set:
+	// `Option<Vec<(T::AccountId, Exposure<T::AccountId, BalanceOf<T>>)>>`
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		/// Just the `Currency::Balance` type; we have this item to allow us to constrain it to
+		/// `From<u64>`.
+		type CurrencyBalance: sp_runtime::traits::AtLeast32BitUnsigned
+			+ codec::FullCodec
+			+ Copy
+			+ MaybeSerializeDeserialize
+			+ core::fmt::Debug
+			+ Default
+			+ From<u64>
+			+ TypeInfo
+			+ Send
+			+ Sync
+			+ MaxEncodedLen;
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		#[pallet::call_index(0)]
+		// #[pallet::weight(T::WeightInfo::new_validators())] // TODO
+		pub fn new_validators(
+			origin: OriginFor<T>,
+			new_validator_set: Vec<(T::AccountId, Exposure<T::AccountId, BalanceOf<T>>)>,
+		) -> DispatchResult {
+			// TODO: origin?
```

Comment:
Should be a permissioned origin that can only be called by XCM. I think from the pallet you can use `EnsureOrigin` and configure it in the runtime. 

Look into the coretime stuff, it uses the same mechanism.